app/controllers/drafts_controller.rb

```diff
@@ -0,0 +1,47 @@
+class DraftsController < ApplicationController
+  requires_login
+
+  skip_before_action :check_xhr, :preload_json
+
+  def index
+    params.require(:username)
+    params.permit(:offset)
+    params.permit(:limit)
+
+    user = fetch_user_from_params
+
+    opts = {
+      user: user,
+      offset: params[:offset],
+      limit: params[:limit]
+    }
+
+    guardian.ensure_can_see_drafts!(user)
+    stream = Draft.stream(opts)
+    stream.each do |d|
+      parsed_data = JSON.parse(d.data) rescue nil
```

Comment:
`rescue nil` is an anti-pattern that we avoid since any and every error is rescued and thrown away. We prefer to rescue the specific error instead.